src/sage/schemes/curves/plane_curve_arrangement.py

```diff
@@ -0,0 +1,1371 @@
+        else:
+            bd = None
+        G, dic = fundamental_group_arrangement(L, simplified=simplified,
+                                               puiseux=True,
+                                               projective=projective,
+                                               vertical=vertical,
+                                               braid_data=bd)
+        if simplified and vertical:
+            self._fundamental_group_simpl_vertical = G
+            self._meridians_simpl_vertical = dic
+        elif simplified and not vertical:
+            self._fundamental_group_simpl_nonvertical = G
+            self._meridians_simpl_nonvertical = dic
+        elif not simplified and vertical:
+            self._fundamental_group_nonsimpl_vertical = G
+            self._meridians_nonsimpl_vertical = dic
+        else:
+            self._fundamental_group_nonsimpl_nonvertical = G
+            self._meridians_nonsimpl_nonvertical = dic
+        return G
+
+    def meridians(self, simplified=True, vertical=True):
+        r"""
+        Return the meridians of each irreducible component.
+
+        OUTPUT:
+
+        A dictionary which associates the index of each curve with its meridians,
+        including the line at infinity if it can be omputed
+
+        .. NOTE::
+
+           This functionality requires the ``sirocco`` package to be installed
+           and :meth:`AffinePlaneCurveArrangements.fundamental_group` with the same options,
+           where some examples are shown.
+
+        EXAMPLES::
+
+            sage: # needs sirocco
+            sage: H.<x, y> = AffinePlaneCurveArrangements(QQ)
+            sage: A = H(x-1, y, x, y - 1)
+            sage: A.fundamental_group()
+            Finitely presented group
+            < x0, x1, x2, x3 | x2*x0*x2^-1*x0^-1, x2*x1*x2^-1*x1^-1,
+                               x3*x0*x3^-1*x0^-1, x3*x1*x3^-1*x1^-1 >
+            sage: A.meridians()
+            {0: [x2], 1: [x0], 2: [x3], 3: [x1], 4: [x3^-1*x2^-1*x1^-1*x0^-1]}
+        """
+        if simplified and vertical:
+            computed = self._meridians_simpl_vertical
+        elif simplified and not vertical:
+            computed = self._meridians_simpl_nonvertical
+        elif not simplified and vertical:
+            computed = self._meridians_nonsimpl_vertical
+        else:
+            computed = self._meridians_nonsimpl_nonvertical
+        if computed:
+            return computed
+        self.fundamental_group(simplified=simplified, vertical=vertical)
+        if simplified and vertical:
+            return self._meridians_simpl_vertical
+        elif simplified and not vertical:
+            return self._meridians_group_simpl_nonvertical
+        elif not simplified and vertical:
+            return self._meridians_nonsimpl_vertical
+        else:
+            return self._meridians_nonsimpl_nonvertical
+
+    def braid_monodromy(self, vertical=True):
```

Comment:
I think this should be an `@cached_method` rather than doing your own caching.

Reply:
In that case, it is true, I would prefer to change it with the other attributes, when `@cached_method(do_pickle=True)` works everywhere